packages/flutter_tools/lib/src/compile.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:usage/uuid/uuid.dart';
+
+import 'artifacts.dart';
+import 'base/file_system.dart';
+import 'base/io.dart';
+import 'globals.dart';
+
+String _dartExecutable() {
+  final String engineDartSdkPath = artifacts.getArtifactPath(
+    Artifact.engineDartSdkPath
+  );
+  return fs.path.join(engineDartSdkPath, 'bin', 'dart');
+}
+
+Future<String> compile({String sdkRoot, String mainPath}) async {
+  final String frontendServer = artifacts.getArtifactPath(
+    Artifact.frontendServerSnapshotForEngineDartSdk
+  );
+
+  if (!sdkRoot.endsWith('/'))
+    sdkRoot = '$sdkRoot/';
+  final Process server = await Process.start(_dartExecutable(),
+    <String>[frontendServer, '--sdk-root', sdkRoot, mainPath]
+  );
+
+  String boundaryKey;
+  String outputFilename;
+  server.stderr
+    .transform(UTF8.decoder)
+    .listen((String s) { print(s); });
```

Comment:
Shorter: `.listen(print)`

Reply:
I actually want to add prefix to all messages, so have to use this more explicit method call(here and below).